### ISSUE TITLE
just wanted to say: This is awesome! 

### ISSUE THREAD

**Issue by ghost** [Lanchon/haystack#2]
Nice work man. Smali is an unreadable b!tch.

**Comment by Lanchon** [comment]
lol thanks!

**Comment by Lanchon** [comment]
btw this is about the best issue i ever got, you should've left it open lol

**Comment by ghost** [comment]
I'm looking to apply this to my stock ROM (Nexus 6P 7.0) which is sadly deodexed, is there a way to deodex only the files that this modifies?

**Comment by Lanchon** [comment]
hey,

i'm a bit against hacking stock. dealing with rooting, mods, otas, and the cat-and-mouse game with the OEM is a pain. what happens in the end is that ppl end up using old unpatched and exploitable OSes.

i'd recommend that you go for fully custom, such as CM, or remain fully stock (no root, nothing).

that said, you have options. you can use latest smali beta (or tool/gui based on it) to deodex the complete rom or just the modded files (service.jar and Settings.apk). only two files, you can even do it manually.

but i'd say don't. you have a very popular and supported device. if you don't want to wipe and go fully custom, why don't you switch to a stock-based recovery flashable distro? you can find them with disabled OTA, pre-rooted, maybe debloated, and even deodexed. and because they are really massaged stock, you can clean-flash /system without wiping /data. picture that the dev you choose will do the whole dance after every OTA, with you just needing to flash.

otherwise you can find the same exact version of modded deodexed flashable stock that you are running and extract the two deodexed files you need (but then you don't learn how to deodex).

finally, you can forgo Haystack altogether and use xposed and the spoof-sign xposed module, if you want seamless updates.

**Comment by ghost** [comment]
I've done custom alot man, just trying stock for a bit. Xposed is also not out for Android N yet

**Comment by Lanchon** [comment]
oh yeah, N! i was using KK for the longest time... until it went unsupported by everybody. that OpenPDroid-patched build was the best android i ever used. i switched to de-googleized MM recently, i guess i'll stick with this one for a long time. i go for MM just in time to see CM dropping it lol :)

**Comment by ghost** [comment]
do you know the commands to deodex with smali.jar?

**Comment by Lanchon** [comment]
no such thing AFAIK. you have to baksmali and then samli the complete jar/apk. baksmali can read odex/oat files.

**Comment by ghost** [comment]
I meant the use of both baksmali & smali, looking to pull the necessary files and if odexed deodex the files.

**Comment by Lanchon** [comment]
i suppose it's quite straightforward. but ive never done it, you'll have to look for the details yourself.

please post your experience here :)

**Comment by Lanchon** [comment]
you can pull with my script. only services.jar and Settings.apk need to be deodexed.

**Comment by Lanchon** [comment]
hey, i gave you wrong info! check this out: https://github.com/JesusFreke/smali/issues/375

> It's worth pointing out that when you change any of the framework jars (e.g. by deodexing them), it invalidates all odexes on the devices, so you have to deodex everything on the system image.

if you use windoze:
http://forum.xda-developers.com/showthread.php?t=2213235

**Comment by ghost** [comment]
@BenBaltz when I reported issues of patching CopperheadOS with the MicroG signature spoofing script, you recommended Haystack instead. Before I go flashing COS on my device, I was curious if root adb is even an option? Or do you have to root your device to toggle root adb? Noob question! Thank you and great project you've created @Lanchon

**Comment by Lanchon** [comment]
this is completely off-topic, please do not post again regarding adb root on this issue. adb root requires a custom kernel or else some kind of huge brittle hack. take care.

**Comment by TheArkive** [comment]
Loving this mod!  Thanks for developing it!

**Comment by IAMtheIAM** [comment]
@Lanchon Thanks from me also! Your DexPatcher/Haystack along with NanoMod helps to allow people like me to regain control of our phones, away from the grips of corporate/govt privacy invading interests. Excellent work.

**Comment by IAMtheIAM** [comment]
Also, one tip that will help others... can you update your readme to tell them that they must be in TWRP Recovery in order for the final  `./push-fileset` command to work. I couldn't figure out why I kept getting `: couldn't create file: Read-only file system`... until someone on a thread somewhere mentioned being inside TWRP recovery to make /System writeable. Rebooted into recovery and then `./push-fileset` worked perfectly. :-D

**Comment by ghost** [comment]
@IAMtheIAM if your rooted you can mount /system rw using terminal emulator

**Comment by Lanchon** [comment]
hi @IAMtheIAM,

thanks for your comments!

golden rule: never overload an issue with stuff about an unrelated issue, no excuse, NEVER. you see that now @BenBaltz responded to you and now whatever this issue was about before (it happened to not be important, this one) is all gone to hell. also, how are people to find info on this new issue if conversation is buried inside a non-related issue? yes, i have a response for you but i refuse to post it here. but i will reply if you create a proper issue, maybe titled as "push-fileset error: couldn't create file: Read-only file system".

thanks again!

**Comment by HawaiiBeach** [comment]
More awesome stuff from @Lanchon! In the future couldn't you make this a executable script for TWRP?

**Comment by Lanchon** [comment]
lol pumpo, this is old as hell !!!! :)

**Comment by IAMtheIAM** [comment]
@Lanchon  Good point about the "golden rule". I'll remember that for future posts :-D

**Comment by SveaMeyer13** [comment]
Sorry for reviving this, but thanks so much.
Honestly, so far I recompiled android for framework changes too complex for smali. This is almost as intuitive but does the job in a matter of minutes. Keep up the amazing work!

**Comment by Tobiaqs** [comment]
Old thread I know, but @Lanchon thanks a lot for developing this!

**Comment by Lanchon** [comment]
thank you so much for the message and your kind attention, few people actually do that. thanks!
